pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -133,4 +133,14 @@ func SetDefaults_NetworkPolicy(obj *extensionsv1beta1.NetworkPolicy) {
 			}
 		}
 	}
+
+	if obj.Spec.PolicyTypes == nil || len(obj.Spec.PolicyTypes) == 0 {
+		// Default for undefined PolicyTypes with defined Egress rules
+		if obj.Spec.Egress != nil || len(obj.Spec.Egress) != 0 {
```

Comment:
I know we discussed changing this yesterday, but I'm starting to doubt my memory.

```
spec:
  egress:
  - to: [ {...} ]
```

Should this policy affect ingress?  I'm thinking it probably shouldn't.

Reply:
Looking back at the email thread where we designed this, I see:

```
  - [ "ingress", "egress" ] if the policy has both ingress: and egress:
    sections

  - [ "egress" ] if the policy has an egress: section but no ingress:
    section

  - [ "ingress"] if the policy has an ingress: section but no egress:
    section, or, for backward compatibility, if it has neither
    ingress: nor egress:.
```

So yeah, my memory was bad.